pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -66,25 +69,30 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
+		r.recorder.Event(instance, corev1.EventTypeWarning, "GetCRD Error", err.Error())
 		return reconcile.Result{}, err
 	}
 
 	// retrieve manifest
 	manifest, err := r.resolver.ResolveCRD(instance.Spec, request.Namespace)
 	if err != nil {
+		r.recorder.Event(instance, corev1.EventTypeWarning, "ResolveCRD Error", err.Error())
 		return reconcile.Result{}, err
 	}
 
 	// TODO validation
 	if len(manifest.InstanceGroups) < 1 {
-		return reconcile.Result{}, fmt.Errorf("manifest is missing instance groups")
+		err = fmt.Errorf("manifest is missing instance groups")
+		r.recorder.Event(instance, corev1.EventTypeWarning, "MissingInstance Error", err.Error())
+		return reconcile.Result{}, err
 	}
 
 	// Define a new Pod object
 	pod := newPodForCR(manifest, request.Namespace)
 
 	// Set BOSHDeployment instance as the owner and controller
 	if err := r.setReference(instance, pod, r.scheme); err != nil {
+		r.recorder.Event(instance, corev1.EventTypeWarning, "NewPodForCR Error", err.Error())
```

Comment:
I don't think any of the 'new pod' events needs a test. The pod functionality is a left over and will probably be removed in the future. 
The BOSHDeployment controller will serve as an entry point, to create the other resources (extended*). But I don't know which story that belongs to and I'm hesitant to remove the newPodCR code and existing tests.

Reply:
I think so.